jfix-zookeeper/src/test/kotlin/ru/fix/zookeeper/lock/PersistentExpiringLockManagerTest.kt

```diff
@@ -66,6 +68,51 @@ internal class PersistentExpiringLockManagerTest {
         callbackFailed.get().shouldBeFalse()
     }
 
+    @Test
+    fun `concurrency releasing and scheduled prolonging lock`() {
+        val manager = createLockManager(PersistentExpiringLockManagerConfig(
+                lockCheckAndProlongInterval = Duration.ofMillis(10)
+        ))
+        val lockIds = mutableListOf<LockIdentity>()
+        for (i in 0 until 100) {
+            val lockId = createLockIdentity()
+            manager.tryAcquire(lockId) { }
+            lockIds.add(lockId)
+        }
+
+        val doneThreadCounter = AtomicInteger(0)
+        val releaseExceptions = mutableListOf<Exception>()
+        val executor = Executors.newFixedThreadPool(Runtime.getRuntime().availableProcessors())
+        for (i in 0 until 50) {
+            val lockId = lockIds[i]
+            executor.execute {
+                try {
+                    TimeUnit.MILLISECONDS.sleep((10 + i).toLong())
+                    manager.release(lockId)
+                } catch (e: Exception) {
+                    releaseExceptions.add(e)
+                } finally {
+                    doneThreadCounter.incrementAndGet()
+                }
+            }
+        }
+        while (doneThreadCounter.get() != 50) {
+            TimeUnit.MILLISECONDS.sleep(100) // wait for releasing all locks
+        }
+        val releaseLockStates = mutableListOf<PersistentExpiringDistributedLock.State>()
+        for (i in 0 until 50) {
+            val lockId = lockIds[i]
+            val stateOptional = manager.getLockState(lockId)
+            if (stateOptional.isPresent) {
+                logger.error("exist state for release lock $lockId state = ${stateOptional.get()}")
+                releaseLockStates.add(stateOptional.get())
+            }
+        }
+
+        releaseExceptions shouldBe emptyList<Exception>()
+        releaseLockStates shouldBe emptyList<PersistentExpiringDistributedLock.State>()
```

Comment:
use explicit assert for collection `.shouldBeEmpty()`

Reply:
fixed